test/e2e/subscription_e2e_test.go

```diff
@@ -1436,22 +1434,17 @@ var _ = Describe("Subscription", func() {
 		require.NoError(GinkgoT(), err)
 		require.NotNil(GinkgoT(), subscription)
 
-		// Check that a single catalog source was used to resolve the InstallPlan
-		_, err = fetchInstallPlan(GinkgoT(), crClient, subscription.Status.InstallPlanRef.Name, buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete))
-		require.NoError(GinkgoT(), err)
-		// Fetch CSVs A, B and C
-		_, err = fetchCSV(GinkgoT(), crClient, csvB.Name, testNamespace, csvSucceededChecker)
-		require.NoError(GinkgoT(), err)
-		_, err = fetchCSV(GinkgoT(), crClient, csvE.Name, testNamespace, csvSucceededChecker)
-		require.NoError(GinkgoT(), err)
-		_, err = fetchCSV(GinkgoT(), crClient, csvA.Name, testNamespace, csvSucceededChecker)
-		require.NoError(GinkgoT(), err)
-		// Ensure csvABC is not installed
-		_, err = crClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), csvABC.Name, metav1.GetOptions{})
-		require.Error(GinkgoT(), err)
-		// Ensure csvD is not installed -- this implies the dependent subscription selected the default channel
-		_, err = crClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), csvD.Name, metav1.GetOptions{})
-		require.Error(GinkgoT(), err)
+		// ensure correct CSVs were picked
+		var got []string
+		Eventually(func() []string {
+			ip, err := crClient.OperatorsV1alpha1().InstallPlans(testNamespace).Get(context.TODO(), subscription.Status.InstallPlanRef.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil
+			}
+			got = ip.Spec.ClusterServiceVersionNames
+			return got
+		}).ShouldNot(BeNil())
+		require.ElementsMatch(GinkgoT(), []string{"nginx-a", "nginx-b-dep", "nginx-e-dep"}, got)
```

Comment:
Nit: Mixture of Gomega matchers and testify assertions, and `Eventually` swallows any errors from the Get.

```suggestion
		Eventually(func() ([]string, error) {
			ip, err := crClient.OperatorsV1alpha1().InstallPlans(testNamespace).Get(context.TODO(), subscription.Status.InstallPlanRef.Name, metav1.GetOptions{})
			if err != nil {
				return nil, err
			}
			return ip.Spec.ClusterServiceVersionNames, nil
		}).Should(ConsistOf("nginx-a", "nginx-b-dep", "nginx-e-dep"))
```